### DISCUSSION TITLE
Game configuration defaults to QWERTY keyboard

### DISCUSSION THREAD

**Discussion by Alastor-git** [voxeltycoon/issues#610] [UI&UX]
**Describe the bug**
The configuration does not take into account the keyboard layout of the user, leading to very tedious setting up. 

**Steps to reproduce**
 - Create a game with non QWERTY keyboard
 - Have to configure everything by hand. 

**Build version**
0.85.0.24 steam

**Platform**
Windows 10

**Comment by katepolak** [comment]
What keys in particular don't work as expected? We will need more information then that becuase there are so many different layouts that it's hard to keep track of.
I am aware of people playing the game on layouts very different from the English QWERTY one so I suspect the issue is with some specific difference with your layout.

**Comment by Alastor-git** [comment]
In my case, for example, I use an AZERTY Keyboard (almost exclusively used in my country). 
Compared to the QWERTY layout, the main differences are : 
 - Invertion of the letters A and Q
 - Invertion of the letters Z and W

As can be seen quickly by looking in the configuration menu, these are amongst the most frequently used keys by the game. So it takes a lot of time to rebind everything and is quite frustrating for a new player. 

Although there are many different keyboard layouts existing, I'd expect the most frequent ones, aka QWERTY, AZERTY, QWERTZ (couldn't find hard facts to support this list, but at the very least that's the ones I know of, that several countries use almost exclusively), to be supported, either automatically or by giving a list of presets to chose from.

**Comment by katepolak** [comment]
I forgot about AZERTY, which is indeed the only mainstream layout that differs significantly from QWERT(Y/Z), the other 2 that are a little niche and mostly used by power users are DVORAK and COLEMAK. The best solution here would likely be to detect the AZERTY layout and apply a different preset upon the first launch, and allow changing the preset in the settings as you suggested.
